paddle/cinn/ir/schedule/ir_schedule.cc

```diff
@@ -120,6 +120,7 @@ class ScheduleImpl {
   void ReverseComputeInline(const Expr& schedule_block);
   void Bind(const Expr& loop, const std::string& thread_axis);
   Expr Rfactor(const Expr& rf_loop, int rf_axis);
+  Expr FactorizeReduction(const Expr& rf_loop, int rf_axis);
```

Comment:
ir_schedule.cc is too long to maintain now ....

I suggest you have following file structure

```
- paddle/cinn/ir/schedule
    - factorize_reduction.h .cc or folder if you also split util files
    - ir_schedule.h .cc  which include factorize_reduction.h
```

Then developers still include `ir_schedule.h` and they can use factorize_reduction, our file is also short.

Reply:
Created a new file factorize_reduction.h and has moved the core implementation of this primitive to this file. However, the interface function FactorizeReduction and function implementation are still preserved in ir_schedule.cc, because this is a member function, and other member functions are called in it.

Reply:
In the future PR, we should still move it.